src/components/viz/unrootedTree/drawUnrootedTree.tsx

```diff
@@ -0,0 +1,233 @@
+import { Node } from "../../../d";
+import * as d3 from "d3";
+//@ts-ignore
+import uuid from "react-uuid";
+import { useSelector } from "react-redux";
+import {
+  getNodeAttr,
+  get_dist,
+  traverse_preorder,
+} from "../../../utils/treeMethods";
+import {
+  containsSamplesOfInterest,
+  getColor,
+} from "../../../utils/unrootedTree";
+
+type DrawUnrootedTreeProps = {
+  colorScale: [string, string, string];
+  chartWidth: number;
+  chartHeight: number;
+  chartMargin: number;
+  scaleDomainX: [number, number];
+  scaleDomainY: [number, number];
+  scaleDomainR: [number, number];
+  tooltip: any;
+};
+
+const outlineColor = "gray";
+
+const tipsRepresented = (node: Node) => {
+  if (node.children.length === 0) {
+    return [node];
+  } else {
+    return node.children.filter(
+      (child: Node) =>
+        child.children.length === 0 && child.branch_attrs.length === 0
+    );
+  }
+};
+
+const drawSquare = (
+  node: Node,
+  mrca: Node,
+  mutsPerTransmissionMax: number,
+  colorScale: [string, string, string],
+  scaleX: Function,
+  scaleY: Function,
+  scaleR: Function,
+  tooltip: any
+) => {
+  const [color, markerColor] = getColor(
+    node,
+    mrca,
+    mutsPerTransmissionMax,
+    colorScale
+  );
+  const size = scaleR(Math.max(5, getNodeAttr(node, "nodeSize")));
+  const x = scaleX(getNodeAttr(node, "x"));
+  const y = scaleY(getNodeAttr(node, "y"));
+
+  return (
+    <g className="square node">
+      <rect
+        x={x - size}
+        y={y - size}
+        width={2 * size}
+        height={2 * size}
+        fill={color}
+        key={`node-${uuid()}`}
```

Comment:
Oh, I don't think this is good. I don't have enough time to think really carefully, but the idea of a `key` in React is to help prevent re-renders when you have a list of many components (usually via `.map` to make components). If each has a unique key determined by its props, then React can re-use that info somehow for less re-renders if the list changes order but maintains the same stuff.

If the key here has `uuid()` being called, then every attempt to re-render would generate a new uuid, so there would not (I'm pretty sure) be any of the benefits React is trying to get, because there would never be constancy around the component keys. This might be able to shut up the `key` warning, but (once again, I'm pretty sure) it's a way to avoid the warning, not the underlying reason the warning is there.